.github/workflows/build_release.yml

```diff
@@ -0,0 +1,94 @@
+name: Cedar-agent and Docker Build
+
+on:
+  pull_request:
+    branches:
+      - main
+  release:
+    types: [published]
+
+jobs:
+  # On PR: Run Cargo tests, build, and update
+  pr_checks:
+    if: github.event_name == 'pull_request'
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Set up Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          profile: minimal
+          override: true
+
+      - name: Run cargo update
+        run: cargo update
+
+      - name: Run cargo test
+        run: cargo test
+
+      - name: Run cargo build
+        run: cargo build
+          
+  # On Release: Publish Cargo package, build and push Docker image
+  release:
+    if: github.event_name == 'release'
+    runs-on: ubuntu-latest
+    timeout-minutes: 70
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Set up Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          profile: minimal
+          override: true
+
+      - name: Update Cargo.toml version
+        run: |
+          sed -i '/\[package\]/,/^version = /s/^version = .*/version = \"${{ github.event.release.tag_name }}\"/' Cargo.toml
+          cat Cargo.toml
+
+      - name: Dry run cargo publish
+        run: cargo publish --dry-run --allow-dirty
+
+      - name: Publishing Rust build artifacts
+        uses: rust-build/rust-build.action@v1.4.5
```

Comment:
Is this a cross compile for both Linux/and and arm ?
The end goal we would like to have here is one binary for each one of linux/amd64 and linux/arm64

Reply:
The supported targets are:
x86_64-pc-windows-gnu
x86_64-unknown-linux-musl
wasm32-wasi
x86_64-apple-darwin

Which mean that I do not see an option to make it for arm.
The other way would be to try and make it manually - I'll try it today